File: src/main/java/com/goudourasv/data/institutions/InstitutionEntity.java
package com.goudourasv.data.institutions;

import com.goudourasv.data.courses.CourseEntity;
import com.goudourasv.data.instructors.InstructorEntity;

import javax.persistence.*;
import java.util.List;
import java.util.Set;
import java.util.UUID;

@Entity(name = "Institutions")
public class InstitutionEntity {
    @Id
    @GeneratedValue
    private UUID id;
    @Column
    private String name;
    @Column
    private String city;
    @Column
    private String country;

    @OneToMany(mappedBy="institutionEntity",cascade = CascadeType.ALL,orphanRemoval = true)
    private List<CourseEntity> courseEntities;

    @ManyToMany(mappedBy = "institutionEntities")
    private  List<InstructorEntity> instructorEntities;



    public UUID getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getCity() {
        return city;
    }

    public String getCountry() {
        return country;
    }

    public List<CourseEntity> getCourseEntities() {
        return courseEntities;
    }

    public List<InstructorEntity> getInstructorEntities() {
        return instructorEntities;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public void setName(String name) {
        this.name = name;
    }

    public void setCity(String city) {
        this.city = city;
    }

    public void setCountry(String country) {
        this.country = country;
    }

    public void setCourseEntities(List<CourseEntity> courseEntities) {
        this.courseEntities = courseEntities;
    }

    public void setInstructorEntities(List<InstructorEntity> instructorEntities) {
        this.instructorEntities = instructorEntities;
    }
}

File: src/main/java/com/goudourasv/data/institutions/InstitutionEntity.kt
package com.goudourasv.data.institutions

import com.goudourasv.data.courses.CourseEntity
import com.goudourasv.data.instructors.InstructorEntity
import java.util.*
import javax.persistence.*

@Entity(name = "Institutions")
class InstitutionEntity(
    @Id
    @GeneratedValue
    var id: UUID? = null,
    @Column(name = "name", nullable = false)
    var name: String,
    @Column(name = "city")
    var city: String? = null,
    @Column(name = "country")
    var country: String? = null,
    //TODO check why in kotlin CascadeType.ALL needs wrap with []
    @OneToMany(mappedBy = "institutionEntity", cascade = ([CascadeType.ALL]), orphanRemoval = true)
    var courseEntities: MutableList<CourseEntity>? = null,
    @ManyToMany(mappedBy = "institutionEntities")
    var instructorEntities: MutableSet<InstructorEntity>? = null,
)



